src/Microsoft.Health.Fhir.CosmosDb/Features/Search/Queries/QueryBuilder.cs

```diff
@@ -71,6 +90,11 @@ public SqlQuerySpec BuildSqlQuerySpec(SearchOptions searchOptions)
                 return query;
             }
 
+            private static string GetCompartmentIndicesParamName(CompartmentType compartmentType)
+            {
+                return $"{KnownResourceWrapperProperties.CompartmentIndices}.{s_compartmentTypeToParamName[compartmentType]}";
```

Comment:
nit: can we add debug check or something to make sure all of the compartment type has a corresponding name entry? Better than failing with generic ArgumentException.